src/main/scala/org/hammerlab/guacamole/reads/MappedRead.scala

```diff
@@ -102,6 +102,63 @@ case class MappedRead(
     (pos, element) => pos + element.getLength
   })
 
+  /**
+   * Take a subsequence or slice of a read [from, until)
+   * Returns None if the provided coordinates do not overlap the read
+   *
+   * @param from reference locus to start the new read
+   * @param until reference locus to slice up to (exclusive end)
+   * @param referenceContigSequence reference sequence for the read's contig
+   * @return A read which spans [from, until) or None if the provided coordinates do not overlap the read
+   */
+  def slice(from: Long, until: Long, referenceContigSequence: ContigSequence): Option[MappedRead] = {
+    if (from >= end || until < start) {
+      None
```

Comment:
Yea, thought for a while on that. I didn't think a 0-length read had any meaning really though so returning None would be cleaner? On a standard sequence it's easy to check for an empty sequence as well (essentially acting as null/None) but here the read the would still exist but its main info (the sequence) would be null.